File: Tyche.Android/pool/src/main/java/com/felipearpa/pool/application/GetPoolGamblerScoresByGamblerUseCase.kt
package com.felipearpa.pool.application

import com.felipearpa.pool.domain.PoolGamblerScoreRepository
import javax.inject.Inject

class GetPoolGamblerScoresByGamblerUseCase
@Inject constructor(private val poolGamblerScoreRepository: PoolGamblerScoreRepository) {

    suspend fun execute(gamblerId: String, next: String? = null, searchText: String? = null) =
        poolGamblerScoreRepository.getPoolGamblerScoresByGambler(
            gamblerId = gamblerId,
            next = next,
            searchText = searchText
        )
}

File: iOS/DataPool/Sources/DataPool/Application/GetPoolGamblerScoresByGamblerUseCase.swift
import Core

public class GetPoolGamblerScoresByGamblerUseCase {
    private let poolGamblerScoreRepository: PoolGamblerScoreRepository
    
    public init(poolGamblerScoreRepository: PoolGamblerScoreRepository) {
        self.poolGamblerScoreRepository = poolGamblerScoreRepository
    }
    
    public func execute(
        gamblerId: String,
        next: String? = nil,
        searchText: String? = nil
    ) async -> Result<CursorPage<PoolGamblerScore>, Error> {
        return await poolGamblerScoreRepository.getPoolGamblerScoresByGambler(
            gamblerId: gamblerId,
            next: next,
            searchText: searchText
        )
    }
}
